File: MultiLineSearch/SearchReplace/ISearchReplaceProvider.vb

Namespace SearchReplace

    ''' <summary>
    ''' Provides search and replace functionality.
    ''' </summary>
    ''' <remarks></remarks>
    Public Interface ISearchReplaceProvider

        ''' <summary>
        ''' Execute search and replace operation.
        ''' </summary>
        ''' <param name="searchKind"></param>
        ''' <param name="findText">Plain text, can contain newlines.</param>
        ''' <param name="replaceText">Plain text, can contain newlines.</param>
        ''' <remarks></remarks>
        Sub ExecSearchReplace(ByVal searchKind As FindReplaceKind, ByVal findText As String, ByVal replaceText As String)

    End Interface

End Namespace

File: MultiLineSearch.Shared/SearchReplace/ISearchReplaceProvider.cs

namespace Helixoft.MultiLineSearch.SearchReplace
{

    /// <summary>
    /// Provides search and replace functionality.
    /// </summary>
    /// <remarks></remarks>
    public interface ISearchReplaceProvider
    {

        /// <summary>
        /// Execute search and replace operation.
        /// </summary>
        /// <param name="searchOptions">Search and replace options.</param>
        /// <param name="findText">Plain text, can contain newlines.</param>
        /// <param name="replaceText">Plain text, can contain newlines.</param>
        /// <remarks></remarks>

        void ExecSearchReplace(FindReplaceOptions searchOptions, string findText, string replaceText);
    }

}